src/java.base/share/classes/jdk/internal/classfile/impl/AbstractInstruction.java

```diff
@@ -314,6 +317,9 @@ static int size(CodeImpl code, int codeStart, int pos) {
             int pad = ap - (pos + 1);
             int low = code.classReader.readInt(ap + 4);
             int high = code.classReader.readInt(ap + 8);
+            if (high < low || high - low > code.codeLength >> 2) {
```

Comment:
May be its also an opportunity to reduce duplication e.g., replace line 316 with a call to `afterPadding()`

Reply:
`BoundTableSwitchInstruction​::afterPadding()` is an instance method, and `BoundTableSwitchInstruction​::size(…)` is a static method, so this would require further refactoring.